File: PrimeNumbers.cpp
#include "stdafx.h"
#include "PrimeNumbers.h"


PrimeNumbers::PrimeNumbers(unsigned first, unsigned last) {
	start = first;
	stop = last;
	// STEP 1:
	progress = 9;

	list_of_primes = { 2, 3, 5, 7 };

	square_root = 0;
}

/**
* Garners the prime numbers between the requested range.
* @param - Nil
*/
string PrimeNumbers::getPrimes() {
	// STEP 2:
	for (; progress < stop; progress += 2) {
		
		do_next_iteration = false; // a flag

		// STEPS 3, 4:
		// Check through already accumulated prime numbers
		// to see if any is a factor of "progress".
		square_root = (int)ceil(sqrt(progress));

		index = 0;
		for (; list_of_primes[index] <= square_root; index++) {
			if ((progress % list_of_primes[index]) == 0) {
				do_next_iteration = true;
				break;
			}
		}
		if (do_next_iteration) {
			continue;
		}

		// if all checks are scaled,then "progress" is our guy.
		list_of_primes.push_back(progress);
	}

	list_of_primes.shrink_to_fit(); // not altogether necessary

	// Display result.
	aux << start;
	result = "The set of prime numbers between " + aux.str() + " and ";
	aux.str("");
	aux << stop;
	result += aux.str() + " are: \n";

	for (unsigned n : list_of_primes) {
		aux.str("");
		aux << n;
		result += aux.str() + "; ";
	}
	return result;
}


PrimeNumbers::~PrimeNumbers() {
}

File: PrimeNumbers.js
var start = 1
var end = 99;
var progress = 9;
var do_next_iteration = false;
var prime_numbers = []; // This array variable would hold our prime numbers as they get verified.

/*
 * STEP 1: 
 */
prime_numbers[0] = 2;
prime_numbers[1] = 3;
prime_numbers[2] = 5;
prime_numbers[3] = 7;
var square_root = 0;
var index = 0; // index into our array
var current = 3; // Points to the current occupied array position.

/*
 * STEP 2:
 */
for (; progress < end; progress += 2) {

    do_next_iteration = false; // a flag

    //STEPS 3, 4:
    /*
     * Alright, one last check.
     * See to it that no prime number less than variable progress is a factor. 
     */
    square_root = Math.abs(Math.sqrt(progress));

    index = 0;
    for (; prime_numbers[index] <= square_root; index++) {
        if ((progress % prime_numbers[index]) == 0) {
            do_next_iteration = true;
            break;
        }
    }
    if (do_next_iteration) {
        continue;
    }

    /* 
     * If 'progress' cleared all those hurdles, then it is definitely a prime number. 
     * We'll also get our subset of smaller prime numbers from here.
     */
    prime_numbers[++current] = progress;
}